main/ble_server.c

```diff
@@ -521,6 +521,8 @@ static void gatts_profile_event_handler(esp_gatts_cb_event_t event, esp_gatt_if_
             is_connected = false;
             disable_notification();
             esp_ble_gap_start_advertising(&spp_adv_params);
+            // TODO: don't restart just because a bluetooth client disconnects
+            esp_restart();
```

Comment:
@bri3d you a fan of this or is this a terrible idea? we don't exactly have valgrind (to my knowledge) to track down potential memory leaks, and passing buffers in tasks, allocating in one and freeing in another is "tough to be perfect", especially when it's happening at ~150 messages/second (typical UDS logging)

Reply:
You're right that cross-task dynamic allocation is rather failure prone (the "right" way to do this would either be to use some kind of lifecycle annotation / refcounting or not to dynamically allocate at all, MISRA style), but the app isn't _that_ big. I don't hate the idea but if we're seeing something leaky enough to fail I'd think we could hunt it down.